pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -0,0 +1,93 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/guregu/null"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/graph/model"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// CreateSystemIntakeActionRequestEdits creates a new action to request edits on an intake form as part of Admin Actions v2
+func CreateSystemIntakeActionRequestEdits(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeRequestEditsInput,
+) (*models.SystemIntake, error) {
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+	// Set the intake current step to the requested edits step
+	formStepMap := map[model.SystemIntakeFormStep]models.SystemIntakeStep{
+		model.SystemIntakeFormStepInitialRequestForm: models.SystemIntakeStepINITIALFORM,
+		model.SystemIntakeFormStepDraftBusinessCase:  models.SystemIntakeStepDRAFTBIZCASE,
+		model.SystemIntakeFormStepFinalBusinessCase:  models.SystemIntakeStepFINALBIZCASE,
+	}
+	intake.Step = formStepMap[input.IntakeFormStep]
+	var targetForm models.GovernanceRequestFeedbackTargetForm
+	// Set the state of the requested form step and set the targeted feedback step
+	switch input.IntakeFormStep {
+	case model.SystemIntakeFormStepInitialRequestForm:
+		intake.RequestFormState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetIntakeRequest
+	case model.SystemIntakeFormStepDraftBusinessCase:
+		intake.DraftBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetDraftBusinessCase
+	case model.SystemIntakeFormStepFinalBusinessCase:
+		intake.FinalBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetFinalBusinessCase
+	default:
+		return nil, &apperrors.BadRequestError{
+			Err: fmt.Errorf("Cannot request edits on %s", input.IntakeFormStep),
+		}
+	}
+	intake, err = store.UpdateSystemIntake(ctx, intake)
+	if err != nil {
+		return nil, err
+	}
+	adminTakingAction, err := fetchUserInfo(ctx, appcontext.Principal(ctx).ID())
+	if err != nil {
+		return nil, err
+	}
+	_, err = store.CreateAction(ctx, &models.Action{
+		ActionType:     models.ActionTypeREQUESTEDITS,
+		ActorName:      adminTakingAction.CommonName,
+		ActorEUAUserID: adminTakingAction.EuaUserID,
+		ActorEmail:     adminTakingAction.Email,
+		BusinessCaseID: intake.BusinessCaseID,
+		IntakeID:       &intake.ID,
+		Feedback:       null.StringFrom(input.EmailFeedback),
+	})
+	if err != nil {
+		return nil, err
+	}
+	govReqFeedback := new(models.GovernanceRequestFeedback)
```

Comment:
Is there any difference in using `new()` here instead of using `&models.GovernanceRequestFeedback{ /* field initialization */ }`? Using `new()` shouldn't be a problem or anything, but we usually use the latter style.

Reply:
OK, I will switch it, thanks.

Reply:
resolved in [c38f771](https://github.com/CMSgov/easi-app/pull/2129/commits/c38f7719b2cd5c17d9a5d3b6c1b2479fef146da5)